### ISSUE TITLE
dynamic formatting doesn't seem to support strtoupper/uppercase

### ISSUE THREAD

**Issue by ANeilan** [openwall/john#5348]
### Checklist ###
- :1st_place_medal: I've read and understood these instructions;
  - This is not a support forum, it's a bug tracker. For questions and support, review postings on the [john-users mailing list](https://www.openwall.com/lists/john-users/).
- :thumbsup: I've tested using latest bleeding version from this repository.
  - Be clear about your environment and what you are doing. Share a sample hash or file that can be used to reproduce.
- :confused: I'm confused and I need guidance.
  - Please, read the instructions at https://www.openwall.com/john/#lists, then join the list before posting.

### IMPORTANT ###
We expect only reports of issues with the latest revision found in this GitHub repository. We do not expect in here, and have little use for, reports of issues only seen in a release or in a distro package.

Attach details about your OS and about john, including:
- The output of `./john --list=build-info`.
[john-build-info.txt](https://github.com/openwall/john/files/12219255/john-build-info.txt)
- The command line you are using. - 
`john/run/john --form=dynamic='md5(md5(strtoupper(md5($p))).$s)' <hash file here>`
or
`john/run/john --form=dynamic='md5(md5(uc(md5($p))).$s)' <hash file here>`

basically, john's dynamic formatting doesn't seem to support strtoupper/uppercasing, leading to the following error message (with either strtoupper or uc):
![image](https://github.com/openwall/john/assets/12475765/f79bce7c-8c8f-43fc-9313-c383316c95b6)

**Comment by ANeilan** [comment]
going to close this, didn't realise that if you uppercase the algorithm, it uppercases the hash
